### PR TITLE
Update links in sundry folders

### PR DESCRIPTION
**Pull request by gewarren**
- replace redirected links
- replace site-relative links with file-relative links

Contributes to #15560

### CODE REVIEW

**Comment by Youssef1313** [comment]
I'd also make the items under `In this section` and `Related section` as bullet list.

**Comment by mairaw** [comment]
@Youssef1313 we've made some changes to our contributor guide recently where we’re trying to be more focused and concentrate in one task per PR:
https://github.com/dotnet/docs/blob/master/CONTRIBUTING.md

File: docs/framework/interop/index.md [hunk 1]
```diff
@@ -34,7 +34,7 @@ Describes marshaling for COM interop and platform invoke.
 [How to: Map HRESULTs and Exceptions](how-to-map-hresults-and-exceptions.md)  
 Describes the mapping between exceptions and HRESULTs.
 
-[COM Wrappers](com-wrappers.md)  
+[COM Wrappers](../../standard/native-interop/com-wrappers.md)
```
Comment (mairaw) [inline docs/framework/interop/index.md]: If this in standard folder, it should not be in the in this section anymore. Move to a Related sections perhaps?

File: docs/framework/net-native/migrating-your-windows-store-app-to-net-native.md [hunk 2]
```diff
@@ -151,7 +151,7 @@ The use of localized resources with the <xref:System.Diagnostics.Tracing.EventSo
 
 **Miscellaneous APIs**
 
-- The [TypeInfo.GUID](xref:System.Type.GUID) property throws a <xref:System.PlatformNotSupportedException> exception if a <xref:System.Runtime.InteropServices.GuidAttribute> attribute isn't applied to the type. The GUID is used primarily for COM support.
+- The <xref:System.Type.GUID?displayProperty=nameWithType> property throws a <xref:System.PlatformNotSupportedException> exception if a <xref:System.Runtime.InteropServices.GuidAttribute> attribute isn't applied to the type. The GUID is used primarily for COM support.
```
Comment (mairaw) [inline docs/framework/net-native/migrating-your-windows-store-app-to-net-native.md]: I’d revert this change. The link text was purposely using a different API name since we can’t represent inherited APIs with traditional links. I’ve opened a bug for that new rule that started showing up.

File: docs/framework/network-programming/handling-errors.md [hunk 3]
```diff
@@ -33,11 +33,12 @@ helpviewer_keywords:
 ms.assetid: 657141cd-5cf5-4fdb-a4b2-4c040eba84b5
 ---
```
Comment (mairaw) [inline docs/framework/network-programming/handling-errors.md]: This title is very generic. Maybe we should open an issue to revisit that.

**Review by mairaw — APPROVED** [review]
Looks good @gewarren. Left some comments for you to consider.

File: docs/framework/network-programming/handling-errors.md [hunk 4]
```diff
@@ -157,11 +158,11 @@ Catch e As Exception
...
+Applications that use the <xref:System.Net.Sockets.Socket> class throw <xref:System.Net.Sockets.SocketException> when errors occur on the Windows socket. The <xref:System.Net.Sockets.TcpClient>, <xref:System.Net.Sockets.TcpListener>, and <xref:System.Net.Sockets.UdpClient> classes are built on top of the **Socket** class and throw **SocketExceptions** as well.  
   
- When a **SocketException** is thrown, the **SocketException** class sets the <xref:System.Net.Sockets.SocketException.ErrorCode%2A> property to the last operating system socket error that occurred. For more information about socket error codes, see the Winsock 2.0 API error code documentation in MSDN.  
+When a **SocketException** is thrown, the **SocketException** class sets the <xref:System.Net.Sockets.SocketException.ErrorCode%2A> property to the last operating system socket error that occurred. For more information about socket error codes, see the Winsock 2.0 API error code documentation in MSDN.  
   
 ## See also
 
-- [Exception Handling Fundamentals](../../standard/exceptions/exception-handling-fundamentals.md)
+- [Exception Handling Fundamentals](../../standard/exceptions/index.md)
```
Comment (mairaw) [inline docs/framework/network-programming/handling-errors.md]:
```suggestion
- [Handling and throwing exceptions in .NET](../../standard/exceptions/index.md)
```

**Comment by gewarren** [comment]
Thanks for the review Maira!
